### PR TITLE
Implement user

### PR DESCRIPTION
**Pull request by at-tuanmai2**

### CODE REVIEW

File: Sources/App/Models/AuthToken.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    // MARK: - Initializing
+    
+    init(row: Row) throws {
+        token = try row.get(AuthToken.Keys.token)
+        userId = try row.get(AuthToken.Keys.userid)
+    }
+
+    func makeRow() throws -> Row {
+        var row = Row()
+        try row.set(AuthToken.Keys.token, token)
```
Comment (at-tamnguyen3) [inline Sources/App/Models/AuthToken.swift]: Please omit `AuthToken`.

File: Sources/App/Models/AuthToken.swift [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        try row.set(AuthToken.Keys.userid, userId)
```
Comment (at-tamnguyen3) [inline Sources/App/Models/AuthToken.swift]: Please omit `AuthToken`.

File: Sources/App/Models/AuthToken.swift [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+// MARK: - JSONConvertible
+// MARK: - Preparation
+
+extension AuthToken: Preparation {
+
+    static func prepare(_ database: Database) throws {
+        try database.create(self) { builder in
+            builder.id()
+            builder.string(AuthToken.Keys.token)
```
Comment (at-tamnguyen3) [inline Sources/App/Models/AuthToken.swift]: Please omit `AuthToken`.

File: Sources/App/Models/AuthToken.swift [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            builder.int(AuthToken.Keys.userid)
```
Comment (at-tamnguyen3) [inline Sources/App/Models/AuthToken.swift]: Please omit `AuthToken`.

File: Sources/App/Models/AuthToken.swift [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    }
+}
+
+// MARK: - Updateable
+
+extension AuthToken: Updateable {
+
+    public static var updateableKeys: [UpdateableKey<AuthToken>] {
+        return [
+            UpdateableKey(AuthToken.Keys.token, String.self) { authToken, token in
```
Comment (at-tamnguyen3) [inline Sources/App/Models/AuthToken.swift]: Please omit `AuthToken`.

File: Sources/App/Models/User.swift [hunk 6]
```diff
@@ -38,6 +46,8 @@ final class User: Model {
     }
     
     init(row: Row) throws {
+        id = try row.get(User.Keys.id)
```
Comment (at-tamnguyen3) [inline Sources/App/Models/User.swift]: Please omit `User`.

File: Sources/App/Routes/UserRoutes.swift [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    var drop: Droplet
+    
+    init(drop: Droplet) {
+        self.drop = drop
+    }
+    
+    func build(_ builder: RouteBuilder) throws {
+        
+        let tokenMiddleware = TokenAuthenticationMiddleware(User.self)
+        let authed = builder.grouped(tokenMiddleware)
```
Comment (at-tamnguyen3) [inline Sources/App/Routes/UserRoutes.swift]: Rename to `auth`.

File: Sources/App/Routes/UserRoutes.swift [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        
+        authed.get("me") { (req) -> ResponseRepresentable in
```
Comment (at-tamnguyen3) [inline Sources/App/Routes/UserRoutes.swift]: Shorten closure into 
```
{ req in 
     <#closure's body#>
}
```

File: Sources/App/Setup/Config+Setup.swift [hunk 9]
```diff
@@ -12,6 +13,8 @@ extension Config {
 
         try setupProviders()
         try setupPreparations()
+        preparations.append(User.self)
```
Comment (at-tamnguyen3) [inline Sources/App/Setup/Config+Setup.swift]: Move into `func setupPreparations()`.

**Comment by at-hanhpham** [comment]
@at-tuanmai2 
Fix conflict

**Comment by at-tuanmai2** [comment]
Xong rồi! 🗡

**Pull request merged**